packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1894,7 +1894,10 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
       final Rect globalRect = MatrixUtils.transformRect(transform, localRect);
```

Comment:
Since we change the selectword contract that it should return next/prev we also need to update handleSelectWord to return pre/next. i.e. if all selectable return next. or if the selectable return prev.

Also should probably assert if a selectable return prev, it must be the first in the selectables.

Reply:
I think I understand why this is needed but for clarification purposes can you confirm my assumptions. We need to update `handleSelectWord` to return `next/prev` because before this change it only return `none/end`. `none` if the word was not selected given the list of selectables managed by the container delegate. And `end` if the word was selected selected given the list of `selectables`.

Now since select word can return `next/prev`, `handleSelectWord` should be updated to also do this in the following cases:
(1) return `next` when the last `SelectionResult` is next
(2) return `prev` when the last `SelectionResult` is prev

Is this correct?

`Also should probably assert if a selectable return prev, it must be the first in the selectables.`
- Not sure I completely understand why for this.

Reply:
> none if the word was not selected given the list of selectables managed by the container delegate. And end if the word was selected selected given the list of selectables.

This is correct prior to this change, though the return value was never used so it doesn't really matter what value is return.

> Now since select word can return next/prev, handleSelectWord should be updated to also do this in the following cases:
(1) return next when the last SelectionResult is next
(2) return prev when the last SelectionResult is prev

(2) is incorrect, handleSelectWord should only return prev if the *first* selectionResult return prev.

 > Is this correct?

yes this is correct and it is necessary because this selectionContainer can be a child selectable of another container, and it can also have the overlap issue with another selectable.

> Not sure I completely understand why for this.

Nvm I think i was wrong, it is possible a selectable that is not the first selectable.

Reply:
Thank you for the clarification, I think I got it!